builtin/credential/ldap/backend.go

```diff
@@ -101,106 +102,191 @@ func (b *backend) Login(req *logical.Request, username string, password string)
 	if c == nil {
 		return nil, logical.ErrorResponse("invalid connection returned from LDAP dial"), nil
 	}
-	binddn := ""
+
+	bindDN, response := getBindDN(cfg, c, username)
```

Comment:
I really like that you split this logic out into different functions, but please follow the convention of returning an `error` rather than returning a `*logical.Response`.